why_missing_maven/why_missing_maven.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+"""Check for missing maven packages and automatically suggest changes in package name correction."""
+# Copy prod bucket:
+#   aws s3 cp s3://prod-keywords-summary prod-keywords-summary --recursive
+#
+# The version file is file containing mapping package-name to package version, e.g.:
+#   'log4j:log4j': '1.0.0'
+#
+# Example run:
+#   why_missing_maven.py -v --no-tagged prod-keywords-summary/maven --version-file maven_latest.json
+
+import json
+import logging
+import os
+import re
+import sys
+from collections import deque
+
+import click
+import daiquiri
+import requests
+import yaml
+
+daiquiri.setup(level=logging.INFO)
+_logger = daiquiri.getLogger(__name__)
+
+
+def iter_files(path):
+    """Yield each file in a directory tree.
```

Comment:
Any particular reason for not using `os.walk`? Would've made this part simpler I think.

Reply:
You can directly pass a directory tree (it will traverse recursively) or a file.